plugins/IntranetMeasurable/Tracker/RequestProcessor.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\IntranetMeasurable\Tracker;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\IntranetMeasurable\Type;
+use Piwik\Tracker\Cache;
+use Piwik\Tracker\Request;
+
+class RequestProcessor extends \Piwik\Tracker\RequestProcessor
+{
+    private $didEnableSetting = false;
+    private $settingName = 'ini.Tracker.trust_visitors_cookies';
+
+    public function manipulateRequest(Request $request)
+    {
+        $site = Cache::getCacheWebsiteAttributes($request->getIdSite());
+        $isIntranetSite = !empty($site['type']) && $site['type'] === Type::ID;
+
+        if ($isIntranetSite && !StaticContainer::get($this->settingName)) {
+            $this->setTrustCookiesSetting(1);
+            $this->didEnableSetting = true;
+        } elseif ($this->didEnableSetting) {
+            // we reset it in case of bulk tracking with different sites etc
+            $this->setTrustCookiesSetting(0);
+            $this->didEnableSetting = true;
```

Comment:
I might be wrong, but I'm not sure this is needed? Could just use an else and unset it if it's not intranet.

Or is it in case another plugin decides to unset it? Actually, won't this ignore the config value then when setting it back?